src/libstd/path.rs

```diff
@@ -2281,8 +2281,8 @@ impl AsRef<OsStr> for Path {
 
 #[stable(feature = "rust1", since = "1.0.0")]
 impl fmt::Debug for Path {
-    fn fmt(&self, formatter: &mut fmt::Formatter) -> Result<(), fmt::Error> {
-        self.inner.fmt(formatter)
+    fn fmt(&self, formatter: &mut fmt::Formatter) -> fmt::Result {
+        fmt::Debug::fmt(&self.inner, formatter)
```

Comment:
I've made several similar changes for these reasons:

* it is easier to visually check that correct trait is called (`Debug` or `Display`)
* to make sure code on Windows and Redox (e. g. `os_str::Slice`) works correctly. I do not have access to Windows machine, so it's easier to write code uniformly (single `fmt` import and explicit trait names) than wait for build system job completion
* and after all to unify all implementations in stack `os_str`, `OsStr`, `Path`